docs/general/ad-filtering/create-own-filters.md

```diff
@@ -1737,6 +1748,212 @@ As a response to blocked request AdGuard returns a short video placeholder.
 >
 > Rules with `$mp4` modifier are not supported by AdGuard Content Blocker, AdGuard for iOS and Safari.
 
+### Rule priorities {#rule-priorities}
+
+Each rule has its own priority, which is necessary when several rules match the request and the filtering system needs to select one of them.
+Priority is measured by a positive integer. The higher the priority of a rule, the higher its priority value.
+In the case of a conflict between two rules with the same priority value, there is undefined behavior, depending on the implementation.
+
+#### Priority counting
+
+To calculate priority, modifiers have been divided into several groups. They are arranged in order of priority from lowest to highest. The more a modifier narrows the scope of a rule, the more priority it adds to the total weight of the rule or the more a rule covers requests, the lower its priority. Note that there are modifiers where setting one parameter has a higher priority than setting two or more, for example, for `$domain=example.com|example.org`, because a rule with two domains has a smaller, infinitesimal, effective area, but still larger than a rule with one resolved domain.
+
+The base priority weight of any rule is 1. If the priority is a floating point number, it will be **rounded up** to the smaller integer greater than or equal to the calculated weight.
+
+> **Compatibility with different versions of AdGuard**
+>
+> Numerical priorities for rules are available in `tsurlfilter` since version `v2.1.0`.
```

Comment:
Does this mean that calculating priorities according to this spec is an optional feature that should be enabled, and the old behaviour should be kept?

Reply:
fixed